src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -130,7 +109,39 @@ public static uint GetResponseHeaderNumberInfo(SafeWinHttpHandle requestHandle,
             return result;
         }
 
-        private static string GetResponseHeaderStringInfo(SafeWinHttpHandle requestHandle, uint infoLevel)
+        // Returns the first header or null if header isn't found.
+        public static string GetResponseHeaderStringInfo(SafeWinHttpHandle requestHandle, uint infoLevel)
+        {
+            uint index = 0;
+            
+            return GetResponseHeaderStringHelper(requestHandle, infoLevel, ref index);
+        }
+
+        // Returns all headers or an empty list if header isn't found.
+        public static List<string> GetResponseHeaders(SafeWinHttpHandle requestHandle, uint infoLevel)
+        {
+            uint index = 0;
+            var headerList = new List<string>();
+            string header;
+            
+            while (true)
+            {
+                header = GetResponseHeaderStringHelper(requestHandle, infoLevel, ref index);
```

Comment:
Good observations. I realize that #3199 is going to optimize some of these memory allocations.  And that pending PR can address this as well. Thx for your patience with that pending PR!

For now, this cookie fix is high priority to address the ASP.NET DNX scenario which is broken right now.